core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -110,19 +112,52 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
 
   private val filteredPayloadSerId = SerializationExtension(system).findSerializerFor(FilteredPayload).identifier
 
+  // key is tuple of entity type and slice
+  private val heartbeatPersistenceIds = new ConcurrentHashMap[(String, Int), String]()
+
+  private def heartbeatPersistenceId(entityType: String, slice: Int): String = {
+    val key = entityType -> slice
+    heartbeatPersistenceIds.get(key) match {
+      case null =>
+        // no need to block other threads, it's just a cache
+        val pid = generateHeartbeatPersistenceId(entityType, slice)
+        heartbeatPersistenceIds.put(key, pid)
+        pid
+      case pid => pid
+    }
+  }
+
+  @tailrec private def generateHeartbeatPersistenceId(entityType: String, slice: Int, n: Int = 1): String = {
+    if (n < 1000000) {
+      val pid = PersistenceId.concat(entityType, s"_ht-$n")
```

Comment:
Should we use `_hb` here, to align with the `HB` for the source name?

Reply:
sure, do you think it is unique enough to not have collision with a user's pid? Otherwise we could make it longer by adding a uuid, either a fixed one or a new one per R2dbcReadJournal instance (same lifecycle as the cache)